skactiveml/tests/template_query_strategy.py

```diff
@@ -630,6 +634,440 @@ def test_query_candidate_variation(self):
+                    replace_query_params=replace_query_params,
+                )
+                y_string = np.full(len(y), "test")
+                test_cases = [(y_string, TypeError)]
+                self._test_param(
+                    "query",
+                    "y",
+                    test_cases,
+                    exclude_clf=True,
+                    replace_query_params=replace_query_params,
+                )
+
+    def test_query_param_candidates(self, test_cases=None):  # TODO more cases
+        test_cases = [] if test_cases is None else test_cases
+
+        for exclude_clf, exclude_reg, query_params in [
+            (False, True, self.query_default_params_clf),
+            (True, False, self.query_default_params_reg),
+        ]:
+            if query_params is not None:
+                ulbd_idx = query_params["candidates"]
+                cases = test_cases + [
+                    (np.nan, ValueError),
+                    (Dummy, ValueError),
+                    ([ulbd_idx[0]], None),
+                ]
+                self._test_param(
+                    "query",
+                    "candidates",
+                    cases,
+                    exclude_clf=exclude_clf,
+                    exclude_reg=exclude_reg,
+                )
+
+    def test_query_param_sample_weight(self, test_cases=None):
+        query_params = inspect.signature(self.qs_class.query).parameters
+        if "sample_weight" in query_params:
+            # custom test cases are not necessary
+            test_cases = [] if test_cases is None else test_cases
+            test_cases += [
+                (np.nan, (ValueError, TypeError)),
+                (Dummy, (ValueError, TypeError)),
+                (None, None),
+            ]
+            self._test_param("query", "sample_weight", test_cases)
+
+            for exclude_clf, exclude_reg, query_params in [
+                (False, True, self.query_default_params_clf),
+                (True, False, self.query_default_params_reg),
+            ]:
+                if query_params is not None:
+                    if not exclude_clf:
+                        replace_query_params = {"fit_clf": True}
+                    else:
+                        replace_query_params = {"fit_reg": True}
+                    y = query_params["y"]
+                    test_cases = [
+                        (np.ones(len(y)), None),
+                        (np.ones(len(y) + 1), ValueError),
+                    ]
+                    self._test_param(
+                        "query",
+                        "sample_weight",
+                        test_cases,
+                        replace_query_params=replace_query_params,
+                        exclude_clf=exclude_clf,
+                        exclude_reg=exclude_reg,
+                    )
+
+    def test_query(
+        self,
+        expected_output,
+        expected_utilities,
+        budget_manager_param_dict=None,
```

Comment:
```suggestion
        budget_manager_param_dict=None,
        X=None,
        y=None,
        candidate=None,
        queried_indices=None
```
Add these parameters as optional ones. If they are set to `None`, we use your current setup as default. As a default case, generating the candidate with a fixed random state would also be better to have better variability.

Reply:
Add comments to have a better overview of this long method.